tests/BaseElementTest.php

```diff
@@ -473,12 +481,16 @@ public function testPreviewLink(string $class, string $elementIdentifier, ?strin
     {
         /** @var BaseElement $element */
         $element = $this->objFromFixture($class, $elementIdentifier);
+        $previewLink = $element->PreviewLink();
 
         if ($link) {
-            $regex = '/^' . preg_quote($link . '?ElementalPreview=', '/') .'\d*#' . $element->getAnchor() . '$/';
-            $this->assertTrue((bool)preg_match($regex, $element->PreviewLink()));
+            $regex = '/^' . preg_quote($link, '/') . '[?&]' . preg_quote('ElementalPreview=', '/') .'\d*#' . $element->getAnchor() . '$/';
+            $this->assertMatchesRegularExpression($regex, $previewLink);
+            // Doesn't try to blindly append query string and anchor - but instead merges intelligently with whatever's already there
+            $this->assertSame(1, substr_count($previewLink, '?'));
+            $this->assertSame(1, substr_count($previewLink, '#'));
```

Comment:
Replaced `assertTrue` with the more correct `assertMatchesRegularExpression` here and modified the regex to account for the new test.

Also counting to make sure we only ever have one query string and one anchor link